### DISCUSSION TITLE
Route color in route planning and navigation

### DISCUSSION THREAD

**Discussion by Leeb64** [devemux86/cruiser#258] [Q&A] [answered]
I have an idea...
When planning, you cannot see the road category (based on the color) through the route line (which covers the road).
If the small route directon arrows were transparent (in planning mode) this would be possible. It would be easier to avoid certain road categories.

**Comment by devemux86** [comment]
@Leeb64 thanks for the interesting idea.

You mean the direction triangles on the route line (enabled in app settings)?

It should be more simple to change their on / off switch to a multiple selection:
- [ ] Routing
- [ ] Navigation

So we can select if we want them disabled or in Routing or Navigation or both.

**Reply by Leeb64** [reply]
I don't want to turn the triangles on or off...
My primary concern is a way of quickly recognizing the road category that is hidden under the route line during route planning.
It would therefore be helpful if the triangles were transparent during planning.

**Reply by devemux86** [reply]
The direction arrows have gap between them, they do not hide the roads.
The route lines may hide the roads if setting a solid color and large width.

The direction arrows can only be turned on or off.

**Comment by Ciutnik** [comment]
> When planning, you cannot see the road category (based on the color) through the route line (which covers the road).  

You can make the route semi-transparent. I've been doing this for years and it's good 🙂

**Reply by Leeb64** [reply]
Only during the planning, that would also be a possibility.
During navigation, in sunlight, semi-transparent would not be optimal...
My route line is therefore black.

**Reply by Leeb64** [reply]
It's only about the planning mode.. A dashed line would also work..
Or something like that

**Reply by devemux86** [reply]
I can add an option for the route color (and transparency) during navigation,
to separate it from route planning.

**Reply by Leeb64** [reply]
That would be good!

**Reply by Ciutnik** [reply]
I know, I used black too. With transparency it became gray - still acceptable. Unfortunately, this requires a compromise. I understand the idea with transparent arrows: they would cut the lines of the route, make triangular holes in it. But I'm afraid those triangles are overlay on overlay.

**Reply by devemux86** [reply]
> But I'm afraid those triangles are overlay on overlay.

Correctly, it is difficult to have good rendering with multiple overlays and mixed transparency.

Let's proceed with the idea of separate color options for route planning and navigation.
We could also have 2 settings for the direction arrows: route planning and navigation?

**Comment by devemux86** [comment]
Different route color settings for route planning and navigation will be available in the next Cruiser:

<img src="https://github.com/devemux86/cruiser/assets/3484020/24339244-f085-4aa6-8f39-13ee8ed5e95f" width="50%"/>

**Reply by 1maxi2** [reply]
Ich bin verwirrt.🤔 Wo gibt es in Cruiser die Möglichkeit schnellste, schnelle und kurvige, kurvige, extra kurvige einzustellen ? Das ist die Funktion die so vermisse da ich  in kurviger alt sehr viel damit geplant habe

**Reply by devemux86** [reply]
@1maxi2 

> Wo gibt es in Cruiser die Möglichkeit schnellste, schnelle und kurvige, kurvige, extra kurvige einzustellen ?

These colors can be useful when you import Kurviger routes.
But with the new BRouter (offline) avoidance options, such route planning is now possible and works **offline**.

Diese Farben können nützlich sein, wenn Sie Kurviger-Routen importieren.
Aber mit den neuen BRouter (offline) Vermeidungsoptionen ist eine solche Routenplanung nun möglich und funktioniert **offline**.

**Reply by DolomitesTom** [reply]
Die Funktion gibt es nicht in Cruiser. Die Einstellung ist nur, falls man eine dermaßen in Kurviger WEB geplante Route in Cruiser importieren würde, bleiben die Farben erhalten. Cruiser funktioniert anders, muss man sich etwas mit beschäftigen, vor allem mit den Routingeinstellungen in Brouter offline. Dafür ist aber, wie der Name schon sagt, alles offline, sowohl die Routenberechnung, als auch die Navigation. Will man die geplante Route so weit wie möglich erhalten, muss man bei der Planung mehr Shapingpunkte setzen oder automatisch beim GPX-Export auffüllen lassen. Beim Import in Cruiser bleibt dann alles erhalten, WPs, SPs, WP-Namen...

**Reply by 1maxi2** [reply]
Schade das es die Möglichkeit zum einstellen der " Kurven" nicht gibt.
Mit den einstellen der Vermeidungen kann man sich helfen aber so einfach ist das nicht weil z.B. das Vermeiden von Hauptstraßen in einer kleinen Ortschaft in Slowenien andere Auswirkungen hat als in einer Großstadt. 
Allerdings ist die offline Planung natürlich ein riesengroßer Vorteil wie erst gestern wieder bei einer 380 km Runde festgestellt !

**Comment by devemux86** [comment, marked as answer]
> Different route color settings for route planning and navigation will be available in the next Cruiser

Available in [Cruiser 3.0.27](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

"Settings | Routes | Appearance | Colors"

**Reply by 1maxi2** [reply]
Hat sich die Kartenansicht in der Planung ( 3.0.27) geändert ? Ich meine die hatte früher anders ausgesehen.
![Screenshot_20230731_141302_Cruiser](https://github.com/devemux86/cruiser/assets/124886589/e2e30a92-b0c4-45cb-9659-a5a1e3d9d927)

**Reply by DolomitesTom** [reply]
ist ein bug bei der Aktivierung von Maps und POIs, schon gemeldet. Farbänderung deaktiviert sowohl Map als auch POIs, ebenso nach jedem Neustart sind die Aktivierungen wieder weg.

**Reply by devemux86** [reply]
@1maxi2 Bitte sehen Sie hier:
- https://github.com/devemux86/cruiser/discussions/262

**Comment by DolomitesTom** [comment]
works fine
